packages/hub/node-tests/utils/scheduled-payments-test.ts

```diff
@@ -9,80 +9,94 @@ describe('ScheduledPaymentUtils', function () {
   // for the same reason (Z is at the end to indicate UTC time).
 
   it('calculates next pay at when current day number is lower than recurring day', function () {
-    let prevPayAt = new Date(Date.parse('2022-02-04T00:00:00.000Z'));
+    let currentDate = new Date(Date.parse('2022-02-04T00:00:00.000Z'));
 
-    let nextPayAt = calculateNextPayAt(prevPayAt, recurringDay, addMonths(prevPayAt, 5));
+    let nextPayAt = calculateNextPayAt(recurringDay, addMonths(currentDate, 5), { currentDate });
     expect(nextPayAt?.toISOString()).to.equal('2022-02-05T00:00:00.000Z');
   });
 
   it('calculates next pay at when current day number is the same as recurring day', function () {
-    let prevPayAt = new Date(Date.parse('2022-02-05T00:00:00.000Z'));
-    let nextPayAt = calculateNextPayAt(prevPayAt, recurringDay, addMonths(prevPayAt, 5));
+    let currentDate = new Date(Date.parse('2022-02-05T00:00:00.000Z'));
+    let nextPayAt = calculateNextPayAt(recurringDay, addMonths(currentDate, 5), { currentDate });
 
-    expect(nextPayAt?.toISOString()).to.equal('2022-03-05T00:00:00.000Z');
+    expect(nextPayAt?.toISOString()).to.equal('2022-02-05T00:00:00.000Z');
```

Comment:
@jurgenwerk based on your suggestion, in this case, the execution payment date is the day of creation.

Reply:
Makes sense